sql/updates/post-update.sql

```diff
@@ -36,6 +36,30 @@ BEGIN
 END
 $$;
 
+--Check for correct upgrade for the caggs with joins
+DO $$
+DECLARE
+  vname regclass;
+  ts_version TEXT;
+BEGIN
+    SELECT extversion INTO ts_version FROM pg_extension WHERE extname = 'timescaledb';
+    IF ts_version >= '2.10.0' THEN
+            CREATE PROCEDURE _timescaledb_internal.post_update_cagg_try_repair(
+                cagg_view REGCLASS
+            ) AS '@MODULE_PATHNAME@', 'ts_cagg_try_repair' LANGUAGE C;
+      FOR vname IN select format('%I.%I', cagg.user_view_schema, cagg.user_view_name)::regclass from _timescaledb_catalog.continuous_agg cagg
+        LOOP
+          SET log_error_verbosity TO VERBOSE;
+          CALL _timescaledb_internal.post_update_cagg_try_repair(vname);
```

Comment:
This is not enough because the current repair function ignores new version of caggs: https://github.com/timescale/timescaledb/blob/main/tsl/src/continuous_aggs/create.c#L2912

And you're not facing problems in your update/downgrade tests because your test case don't contain the problem you're fixing with this PR.